### PR TITLE
Update value specification specs

### PR DESCRIPTION
**Pull request by akphi**
## Summary

- [x] Bump dependencies
- [x] Update `V1_ValueSpecification` per the change in https://github.com/finos/legend-engine/pull/1063:
  - [x] Reorganized
  - [x] Update to use static multiplicities
  - [x] Document about `V1_ClassInstance` and `V1_CoreClassInstance`
  - [x] Updated `V1_ValueSpecification` and `V1_ValueSpecificationVisitor` to support the latest list of value specification
  - [x] Take care of (de)serialization of different `V1_ClassInstance` and introduce extension mechanism for it: e.g. `serialize(V1_mappingModelSchema(this.plugins), element)`
- [x] Remove all explicit support for `V1_HackedUnit` and `V1_HackedClass`, these now should be removed all together and converted to `V1_GenericTypeInstance`.
- [x] Handle CString with value as empty list: follow the change in https://github.dev/finos/legend-engine/pull/1076
- [x] Update `value spec` builder/transform/path-resolver logic, and anything else that implements `V1_ValueSpecificationVisitor`roundtrip test for value specification
- [x] Hard-code all multiplicity for all primitive value specification
  - [x] Remove `multiplicity` field from serialization schema for all of `V1_PrimitiveValueSpecification`
  - [x] Globally use static `Multiplicity` and `V1_Multiplicity`: we should not see
  - [x] Set the type to `Nil` - // for arguments of types we don't support, we will fill them with `[]` - Check location: `return new CollectionInstanceValue(graph.getTypicalMultiplicity(TYPICAL_MULTIPLICITY_TYPE.ZERO));`, we now create an utility method to represent nullish value
- [x] Adjust value specification metamodels:
  - [x] Eradicate the following value specification metamodels and their visitors
        // visit_AlloySerializationConfigInstanceValue(valueSpecification: AlloySerializationConfigInstanceValue): T;
        // visit_RuntimeInstanceValue(valueSpecification: RuntimeInstanceValue): T;
        // visit_PairInstanceValue(valueSpecification: PairInstanceValue): T;
        // visit_MappingInstanceValue(valueSpecification: MappingInstanceValue): T;
        // visit_PureListInstanceValue(valueSpecification: PureListInstanceValue): T;
        // visit_PropertyGraphFetchTreeInstanceValue(PropertyGraphFetchTreeInstanceValue: PureListInstanceValue): T;
  - [x] Rename `RootGraphFetchTreeInstanceValue` to `GraphFetchTreeInstanceValue`
  - [x] Document about why we have different kinds of `InstanceValue`: move them all into one file and document there, we don't have a powerful matcher like in `Pure`, see https://github.com/finos/legend-engine/blob/94cd0aff3b314e568e830773f7200c8245baa09b/legend-engine-pure-code-compiled-core/src/main/resources/core/pure/protocol/vX_X_X/transfers/valueSpecification.pure#L254
- [x] Remove `TYPICAL_MULTIPLICITY_TYPE` and `PureModel.getTypicalMultiplicity()`
- [x] Make deserialization of `included mapping` backward compatible with fields like `includedMappingName` and `includedMappingPackage`
- [x] Leverage `PlainObject` whenever possible to replace all `Record<PropertyKey, unknown>`
- [x] Create roundtrip test for value specification serializer taking into account backward compatibility changes we made
- [x] Make `ModelStore` a singleton
- [x] Make primitive types singletons
- [x] Minor cleanups post #1587 
- [x] improve documentation for `local-development-assemblage` workflow 

## How did you test this change?

- [x] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
great!

File: packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_PersistenceCloud-basic.pure [hunk 1]
```diff
@@ -1,193 +0,0 @@
-###Pure
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_PersistenceCloud-basic.pure:1]: It's duplicated of the other test and it's faulty, i.e. `PersistentContext` is not compiling right now

File: packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_Persistence-basic.pure [hunk 2]
```diff
@@ -177,13 +177,3 @@ Persistence org::legend::ZooPersistence
 
   ]
 }
-
-PersistenceContext org::legend::ZooPersistenceContext
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-manual-tests/src/__tests__/roundtrip-grammar/cases/DSL_Persistence-basic.pure:181]: see above

File: packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/function/V1_ConcreteFunctionDefinition.ts [hunk 3]
```diff
@@ -41,6 +41,18 @@ export class V1_ConcreteFunctionDefinition extends V1_PackageableElement {
    * @discrepancy model
    */
   body: object[] = [];
+  /**
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-graph/src/graphManager/protocol/pure/v1/model/packageableElements/function/V1_ConcreteFunctionDefinition.ts:44]: yes

File: packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureProtocol/serializationHelpers/V1_MappingSerializationHelper.ts [hunk 4]
```diff
@@ -1142,7 +1142,22 @@ export const V1_mappingModelSchema = (
...
+      custom(
+        (val) => serialize(V1_mappingIncludeModelSchema, val),
+        (val) =>
+          deserialize(V1_mappingIncludeModelSchema, {
+            ...val,
+            /**
+             * @backwardCompatibility
+             */
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            includedMapping:
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-graph/src/graphManager/protocol/pure/v1/transformation/pureProtocol/serializationHelpers/V1_MappingSerializationHelper.ts:1155]: wut?

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**